File: test_data/code/python/1588-sum-of-all-odd-length-subarrays.py
class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        n = len(arr)
        total_sum = 0
        for length in range(1, n + 1, 2):
            for i in range(n - length + 1):
                total_sum += sum(arr[i:i + length])
        return total_sum

class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        n = len(arr)
        prefix_sum = [0] * (n + 1)
        for i in range(n):
            prefix_sum[i + 1] = prefix_sum[i] + arr[i]
        total_sum = 0
        for length in range(1, n + 1, 2):
            for i in range(n - length + 1):
                total_sum += prefix_sum[i + length] - prefix_sum[i]
        return total_sum

File: test_data/code/java/1588-sum-of-all-odd-length-subarrays.java
class Solution {
    public int sumOddLengthSubarrays(int[] arr) {
        int n = arr.length;
        int totalSum = 0;

        for (int i = 0; i < n; i++) {
            for (int j = i; j < n; j++) {
                if ((j - i + 1) % 2 != 0) {
                    totalSum += Arrays.stream(arr, i, j + 1).sum();
                }
            }
        }

        return totalSum;
    }
}